gluefactory/utils/tensor.py

```diff
@@ -46,3 +47,79 @@ def index_batch(tensor_dict):
     batch_size = len(next(iter(tensor_dict.values())))
     for i in range(batch_size):
         yield map_tensor(tensor_dict, lambda t: t[i])
+
```

Comment:
I would create an own file in utils, since your methods here are not purely dealing with tensors